# Analysis of *E. coli* resistant to infection by the T5 phage

We will use re-sequencing to understand the molecular nature of *E. coli* resistance to the T5 bacteriophage.

This step uses sequenced raw reads (in compressed `fastq` files) to align them to the bacterial reference genome. Note that there are data on four squenced mutants and the wildtype in the `reads` folder.

## Load the R libraries 

This step incorporates commands not found in the base R language

```{r}
library(glue) # for easily generating command line commands
```

It Visualize the read quality via fastQC <optional>
#Download link-https://www.bioinformatics.babraham.ac.uk/projects/download.html <available for windows and Mac users>
#Check the output file-"mutant1_OIST-2015-03-28_fastqc.html"

## Generate an index for the reference genome

Most aligners require a step like that for their algoritm to work. Genome alignment is not part of R, so we'll be executing a shell command using `system`.

```{r}
system("bowtie2-build ref/NC_012967.1.fasta ref/ecoli")
```

Note that a bunch of files with the extension `.bt2` have been created in the `ref/` folder. That's the index.

## Map the mutant E.coli reads to the reference genome

Like the aligner, the mapper is not part of base R. We'll create a command that maps the samples and sorts them. In this case we're using the 'bowtie2' aligner and `samtools` for converting between file formats, sorting and creating an index for the resulting files. This command is a bit complex, and you don't need to understand all of it. In fact, it's actually five separate commands that pass input to each other. But do know that you can easily change which sample you want to run by changing the `sequencedSample` variable. The `glue` command just inserts this variable into the string we'll execute in the shell.

```{r}
sequencedSample  <- "mutant3_OIST-2015-03-28"
```

```{r}
(bowtieCommand <- glue("bowtie2 -x ref/ecoli reads/{sequencedSample}.fq.gz --rg-id {sequencedSample} --rg SM:{sequencedSample} |
samtools view -Su - | 
samtools sort -o {sequencedSample}.bam - && 
samtools index {sequencedSample}.bam"))
system(bowtieCommand)
```

You will see that a files called `sequencedSample.bam` and its corresponding index `sequencedSample.bam.bai` have been created in the root directory, where `sequencedSample` is the name of the variable you assigned in the this code chunk. You'll need both files for the genome browser later.

What's inside an alignemnt file? You can read all about it in the [SAM format specification](https://samtools.github.io/hts-specs/SAMv1.pdf). Note that this is a file with a `.bam` extension, but it is basically just a compressed `.sam` file.

```{r}
cat(system(glue("samtools view {sequencedSample}.bam | head -5"), intern = T))
```

Ignore the `cat` command and the `intern = T` argument. They exist only to make the output prettier if you convert this document to HTML.

## Variant Calling

Again, we're using a standalone algorithm, in this case the `freebayes` software. The variant caller assigns a [quality score](https://en.wikipedia.org/wiki/Phred_quality_score) to each site, depending on how likely it is to exist in the data. The `vcffilter` gets rid of low-quality sites.

We're cheating a little bit and asking freebayes only to call variants in the vicinity of the FhuA gene, i.e., on chromosome NC_012967.1 between bases 170,000 and 173,000. This speeds things up.

```{r}
(freebayesCommand <- glue("freebayes -r NC_012967.1:170000-173000 -f ref/NC_012967.1.fasta {sequencedSample}.bam -p 1 | 
vcffilter -f \"QUAL > 20\"  > {sequencedSample}.vcf"))

system(freebayesCommand)
```

This will generate a `.vcf` file. You can and probably should take a look at it in a text editor. Looking at just a few columns, you see that there is a set of coordinates (first two columns), the reference base, the variant and its variant quality. I'm omitting a bunch of fields for simplicity.


```{r}
cat(system(glue("grep -m2 -v ^# {sequencedSample}.vcf | cut -f1,2,4,5,6"), intern = T))
```

## Visualize results

You will now have the alignment (files ending with `.bam` and its index `.bam.bai`), and the variant call file (`.vcf`), which are the end points of the computational analysis. You should download them and the reference sequence, index and annotation files (`ref/NC_012967.1.*`) to your local computer.

You will need to download the data from the virtual machine:

<iframe width="560" height="315" src="https://www.youtube.com/embed/-AbSUeibAqI" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

Then you can load the data into the IGV browser:

<iframe width="560" height="315" src="https://www.youtube.com/embed/hhCeJD7HRdI" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>